app/controllers/issues_controller.rb

```diff
@@ -82,7 +83,14 @@ def update
         @modified = true
         check_for_edit_conflicts(@issue, updated_at_before_save)
         track_updated(@issue)
-        format.html { redirect_to project_issue_path(current_project, @issue), notice: 'Issue updated' }
+        format.html do
+          redirect_path = if params[:redirect_to] && URI.parse(params[:redirect_to]).relative?
```

Comment:
Instead of calling `params[:redirect_to] && URI.parse(params[:redirect_to]).relative?` multiple times, can we use a before_action cal instead? Something like:

```
def set_redirect_to
  @redirect_to =
    if URI.parse(params[:redirect_to]).relative?
      params[:redirect_to]
    else
      raise 'Something is fishy...'
    end
end
```

Then we can be sure `@redirect_to` is a local path and use it where we need to

Reply:
This has actually changed in the #edit view PR and it's only called once in the controller. 

Leaving this as is here to prevent a merge conflict when the #edit view PR is merged into this PR.

Reply:
First I thought, Rails 7 cuts this by default and requires :allow_other_host if you want to go out of your way. That seems also to be available in our 6.1

https://github.com/rails/rails/blob/6-1-stable/actionpack/lib/action_controller/metal/redirecting.rb

But then I also remember `#redirect_back` couldn't we generally use that to avoid having to manually handle this?

Reply:
If we use `#redirect_back` the user will submit the form and land back in #edit, since #update will redirect_back to #edit.

I tested the redirect without `URI.parse(params[:redirect_to]).relative?` and it seems like rails will allow the redirect to another host 

Instead of using `params[:redirect_to]`, we could approach this with `session[:return_to]` instead but regardless, I don't see a reliable solution without passing in the path to `session[:return_to]` as we do now with `params[:redirect_to]`

Alternatively, we can check `request.referrer` in the controller and set the redirect path based on that. This should eliminate the need for `params[:redirect_to]`

Thoughts? cc @aapomm